vunit/project.py

```diff
@@ -530,7 +530,7 @@ def add_source_file(self, source_file):
         """
         if source_file.name in self._source_files:
             old_source_file = self._source_files[source_file.name]
-            if old_source_file.content_hash != source_file.content_hash:
+            if old_source_file.file_hash != source_file.file_hash:
```

Comment:
For Verilog this was not equivalent, duplicate files could be added with different defines or include paths. The solution you implemented would silently ignore that a verilog file was added twice with different defines. This was caught earlier since content_hash included a hash of the defines. Maybe there should be a unit test that duplicate verilog files with different defines fail. If that test existed this PR would fail the test.